### DISCUSSION TITLE
Integrating Ory Kratos and Oathkeeper for Native Apps (Mobile)

### DISCUSSION THREAD

**Discussion by susanto-tm** [ory/oathkeeper#887] [Q&A] [answered]
Hi, I’ve just started playing around with Ory Kratos and Oathkeeper. I’ve been looking into using Kratos for a React Native app and Oathkeeper in front of my private APIs. I was able to get Oathkeeper to work using the Kratos endpoints for the browser by using the Ory Session Cookie, but I am confused about how to get Oathkeeper to work with the API flow with Kratos. It seems that I could use the Ory Session Token as a bearer token to verify with the /whoami endpoint but I keep hitting an Unauthorized response. Any help about integrating the two together for a mobile app would be appreciated.

**Comment by aeneasr** [comment, marked as answer]
Yes, you could use the check session endpoint to check the token at ory kratos‘ whoami endpoint, similar to how you do it for sessions :)
